### PR TITLE
Add `incus file create` subcommand

### PR DESCRIPTION
**Pull request by monstermunchkin**
This adds a new `incus file create` subcommand which allows creating files,
symlinks, and directories without having to specify a source.

### CODE REVIEW

**Comment by stgraber** [comment]
@monstermunchkin oops, I just merged a Japanese translation update and that's causing a conflict, can you do a quick rebase?

File: cmd/incus/file.go [hunk 1]
```diff
@@ -96,16 +108,214 @@ func (c *cmdFile) Command() *cobra.Command {
...
+	cmd := &cobra.Command{}
+	cmd.Use = usage("create", i18n.G("[<remote>:]<instance>/<path> [<symlink target path>]"))
+	cmd.Short = i18n.G("Create files and directories in instances")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Create files and directories in instances`))
+	cmd.Example = cli.FormatSection("", i18n.G(
+		`lxc file create foo/bar
+	   To create a file /bar in the foo instance.
+lxc file create --type=symlink foo/bar baz
+	   To create a symlink /bar in instance foo whose target is baz.`))
```
Comment (stgraber) [inline cmd/incus/file.go:138]: Pretty sure that should say `incus` :)

File: cmd/incus/file.go [hunk 2]
```diff
@@ -30,8 +30,12 @@ import (
 	"github.com/lxc/incus/shared/util"
 )
 
-// DirMode represents the file mode for creating dirs on `incus file pull/push`.
-const DirMode = 0755
+const (
+	// DirMode represents the file mode for creating dirs on `incus file pull/push`.
+	DirMode = 0755
+	// FileMode represents the file mode for creating files on `lxc file create`.
```
Comment (stgraber) [inline cmd/incus/file.go:36]: `incus`

File: cmd/incus/file.go (hunk 1 above)
Comment (monstermunchkin) [reply cmd/incus/file.go]: Fixed.

File: cmd/incus/file.go (hunk 2 above)
Comment (monstermunchkin) [reply cmd/incus/file.go]: Fixed.

**Comment by monstermunchkin** [comment]
> @monstermunchkin oops, I just merged a Japanese translation update and that's causing a conflict, can you do a quick rebase?

Done.

**Comment by stgraber** [comment]
@monstermunchkin cool, so that makes sense, though I'm not sure about the `--content` and stdin stuff. If you want to actually send through a file or file content, you probably should just use `incus file push` which already can read from stdin (so you can always `echo content | incus file - push foo/tmp/file`

**Comment by monstermunchkin** [comment]
> @monstermunchkin cool, so that makes sense, though I'm not sure about the `--content` and stdin stuff. If you want to actually send through a file or file content, you probably should just use `incus file push` which already can read from stdin (so you can always `echo content | incus file - push foo/tmp/file`

OK, I'll remove that part then.

**Review by stgraber — APPROVED** [review]

**Pull request merged**